src/SSW.Rewards.Application/Quizzes/Commands/AddNewQuiz/AddNewQuizCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿using SSW.Rewards.Application.Achievements.Common;
+using SSW.Rewards.Application.Quizzes.Common;
+using SSW.Rewards.Domain.Enums;
+
+namespace SSW.Rewards.Application.Quizzes.Commands.AddNewQuiz;
+public class AddNewQuizCommand : IRequest<int>
+{
+    public QuizDetailsDto Quiz { get; set; }
+
+    public int Points { get; set; }
+
+    public Icons Icon { get; set; }
+}
+
+public class AddNewQuizCommandHandler : IRequestHandler<AddNewQuizCommand, int>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly ICurrentUserService _currentUserService;
+    private readonly IUserService _userService;
+
+    public AddNewQuizCommandHandler(
+        IApplicationDbContext context,
+        ICurrentUserService currentUserService,
+        IUserService userService)
+    {
+        _context = context;
+        _currentUserService = currentUserService;
+        _userService = userService;
+    }
+
+    public async Task<int> Handle(AddNewQuizCommand request, CancellationToken cancellationToken)
+    {
+        var user = _userService.GetCurrentUser();
+
+        var dbUser = await _context.Users.FindAsync(user.Id, cancellationToken);
+
+        var quiz = new Quiz
+        {
+            Title       = request.Quiz.Title,
+            Description = request.Quiz.Description,
+            Icon        = request.Icon,
+            IsArchived  = false,
+            CreatedBy   = dbUser
+        };
+
+        foreach (var question in request.Quiz.Questions)
+        {
+            var dbQuestion = await _context.QuizQuestions.FirstOrDefaultAsync(q => q.Id == question.QuestionId, cancellationToken);
+
+            if (dbQuestion is null)
+            {
+                dbQuestion = new QuizQuestion
+                {
+                    Text = question.Text
+                };
```

Comment:
Move this out of the if(... as a result of the above comment

Reply:
Done